cmd/oras/internal/option/confirmation.go

```diff
@@ -45,7 +45,12 @@ func (opts *Confirmation) AskForConfirmation(r io.Reader, prompt string) (bool,
 
 	scanner := bufio.NewScanner(r)
 	if ok := scanner.Scan(); !ok {
-		return false, scanner.Err()
+		err := scanner.Err()
```

Comment:
Code changes here covers this case: when scanner fail with EOF error, the returned error will be nil without any logging.